stdlib/json.gr

```diff
@@ -0,0 +1,2155 @@
+    }
+  } else {
+    // Special handling for the escapeHTMLUnsafeSequences flag.
+    // Escaping a sequence requires keeping track of previous characters,
+    // which is difficult and suboptimal when using a function to iterate
+    // the input string. So we don't want to pay the price in other cases.
+    // This cannot be done just in the emitCodePoint function.
+    // It could be possible to implement more optimally, but would
+    // complicate things even more than this.
+    (s: String) => {
+      Buffer.addChar('"', buffer)
+
+      let mut prevCodePoint = 0
+
+      String.forEachCodePoint(codePoint => {
+        if (codePoint == 47) {
+          if (format.escapeHTMLUnsafeSequences && prevCodePoint == 60) {
+            Buffer.addChar('\\', buffer)
+            Buffer.addChar('/', buffer)
+          } else {
+            // otherwise just emit the slash as-is
+            addCharFromCodePoint(codePoint, buffer)
+          }
+        } else {
+          emitCodePoint(codePoint)
+        }
+
+        prevCodePoint = codePoint
+      }, s)
+
+      Buffer.addChar('"', buffer)
+    }
+  }
+
+  let implHelper = {
+    format,
+    bufferPretty: buffer,
+    emitEscapedQuotedStringPretty: s => {
+      emitEscapedQuotedString(s)
+    },
+    printNewLine,
+    printIndentation,
+  }: JsonWriterPrettyImplHelper
+
+  {
+    emit: json => {
+      let error = printElementPretty(json, implHelper, 0)
+      if (Option.isSome(error)) return error
+      if (format.finishWithNewLine) printNewLine()
+      return None
+    },
+  }: JsonWriter
+}
+
+/**
+ * Prints the JSON object into a string with specific formatting settings.
+ *
+ * @param json: The JSON object to print
+ * @param format: formatting option
+ * @returns A `Result` object with either the string containing the printed JSON or an error if the input object cannot be represented in the JSON format.
+ *
+ * @example
+ * print(
+ *   Result.unwrap(
+ *     toString(
+ *       format=Custom(defaultCompactAndSafeFormat),
+ *       JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *     )
+ *   )
+ * )
+ * // Output: {"currency":"\u20ac","price":99.9}"
+ * @example
+ * print(
+ *   Result.unwrap(
+ *     toString(
+ *       JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))]
+ *     )
+ *   )
+ * )
+ * // Output: {"currency":"€","price":99.9}
+ * @example
+ * print(
+ *   Result.unwrap(
+ *     toString(
+ *       format=Compact
+ *       JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *      )
+ *   )
+ * )
+ * // Output: {"currency":"€","price":99.9}
+ * @example
+ * print(
+ *   Result.unwrap(
+ *     toString(
+ *       format=Pretty,
+ *       JsonObject([("currency", JsonString("€")), ("price", JsonNumber(99.9))])
+ *     )
+ *   )
+ * )
+ * // Output:
+ * // {
+ * // "currency": "€",
+ * // "price": 99.9
+ * //}
+ *
+ * @since v0.6.0
+ */
+provide let toString = (format=Compact, json: Json) => {
```

Comment:
So `Compact` is the default... we need tests for using `toString` without a specified format. And it seems that we also don't have tests for `Pretty` 🤔